### PR TITLE
New OS: Adva XG 304

### PR DESCRIPTION
**Pull request by h-barnhart**
This PR introduces the Adva XG300 series to LibreNMS, specifically the XG304 model variant. Includes basic device identification and information, port data, optical device health graphs, and device temperature sensors. Haven't been able to capture any meaningful traps yet, I will probably do that later.

The Adva XG300 series is based on an acquisition of MRV Networking, so the OS is completely different from the other Adva devices currently in LibreNMS. Under the hood its Linux with a Cisco IOS-like CLI on top. MIBs are bit of MRV, generic, and Adva. I think I've only added the MIBs required for discovery and polling of the device as required for the ports and sensors in this PR.

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [x] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: LibreNMS/OS/AdvaXg300.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+namespace LibreNMS\OS;
+
+use App\Models\Device;
+use LibreNMS\Interfaces\Discovery\OSDiscovery;
+use LibreNMS\OS;
+
+class AdvaXg300 extends OS implements OSDiscovery
+{
+    public function discoverOS(Device $device): void
+    {
```
Comment (Jellyfrog) [inline LibreNMS/OS/AdvaXg300.php:35]: Would you mind implementing this in yaml instead? You can use regex to get the fields

**Review by PipoCanaja — DISMISSED** [review]
LGTM

**Comment by laf** [comment]
This might have changed since I was previously involved in LIbreNMS but OS' are usually not device range specific, do the Adva XG 300 range use a firmware/os that doesn't exist on any other XG ranges?

**Comment by h-barnhart** [comment]
> This might have changed since I was previously involved in LIbreNMS but OS' are usually not device range specific, do the Adva XG 300 range use a firmware/os that doesn't exist on any other XG ranges?

Yes, the XG300 series is different. In the current version of LibreNMS the XG300 series is discovered as a generic device, it is neither identified as Adva or MRV.

The XG100 and XG200 series were developed by Adva and run their in house OS which has a usable web UI and an "interesting" CLI. The XG300 series comes from an acquisition of MRV Networks and runs their OS. The CLI is IOS like and there is no a web UI.

File: includes/definitions/discovery/adva_xg300.yaml [hunk 2]
```diff
@@ -0,0 +1,96 @@
+mib: OA-SFP-MIB:DEV-CFG-MIB
+modules:
+    os:
+        sysDescr_regex: '/(?<hardware>FSP150-XG3\d+\s\w+\s\w+-\w) Ver\. (?<version>[\d+_]+\d \(\d+\)$)/'
```
Comment (Jellyfrog) [inline includes/definitions/discovery/adva_xg300.yaml:4]: Maybe nitpicking but I'd suggest using something a bit more ... forgiving:
```suggestion
        sysDescr_regex: '/(?<hardware>.+) Ver\. (?<version>[^)]+\))/'
```

Comment (h-barnhart) [reply includes/definitions/discovery/adva_xg300.yaml]: Ok, I could've clicked a button and fixed that. Oh well, I put it in.

**Pull request merged**